src/config.rs

```diff
@@ -443,106 +544,72 @@ impl Cfg {
             .transpose()?)
     }
 
-    pub(crate) fn find_override(
+    pub(crate) fn find_active_toolchain(
         &self,
         path: &Path,
-    ) -> Result<Option<(LocalToolchainName, OverrideReason)>> {
-        Ok(self
-            .find_override_config(path)?
-            .and_then(|(override_cfg, reason)| override_cfg.toolchain.map(|t| (t, reason))))
-    }
-
-    fn find_override_config(&self, path: &Path) -> Result<Option<(OverrideCfg, OverrideReason)>> {
-        let mut override_ = None;
-
-        // First check toolchain override from command
-        if let Some(ref name) = self.toolchain_override {
-            override_ = Some((name.to_string().into(), OverrideReason::CommandLine));
-        }
-
-        // Check RUSTUP_TOOLCHAIN
-        if let Some(ref name) = self.env_override {
-            // Because path based toolchain files exist, this has to support
-            // custom, distributable, and absolute path toolchains otherwise
-            // rustup's export of a RUSTUP_TOOLCHAIN when running a process will
-            // error when a nested rustup invocation occurs
-            override_ = Some((name.to_string().into(), OverrideReason::Environment));
-        }
-
-        // Then walk up the directory tree from 'path' looking for either the
-        // directory in override database, or a `rust-toolchain` file.
-        if override_.is_none() {
-            self.settings_file.with(|s| {
-                override_ = self.find_override_from_dir_walk(path, s)?;
-
-                Ok(())
-            })?;
+    ) -> Result<Option<(LocalToolchainName, ActiveReason)>> {
+        if let Some((override_config, reason)) = self.find_override_config(path)? {
```

Comment:
How about using `Ok(if let ...)` here instead of `if let ... { Ok(...) } else { Ok(...) }`?